docs/data/date-pickers/date-calendar/ReverseYearCalendar.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import dayjs from 'dayjs';
+import { DemoContainer, DemoItem } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { YearCalendar } from '@mui/x-date-pickers/YearCalendar';
+
+const currentYear = dayjs();
+
+export default function ReverseYearCalendar() {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <DemoContainer components={['YearCalendar']}>
+        <DemoItem label="Years in reverse">
```

Comment:
```suggestion
        <DemoItem label="Years in reversed order">
```

Or

```suggestion
        <DemoItem label="Years in descending order">
```

Which makes me think that a different and maybe more explicit API could have been `yearsOrder?: 'asc' | 'desc'`.
If at some point we change the default order, I guess we don't want people that use `reverseYears` to switch to `asc` :laughing: 

But given the amount of work already done here, I don't want to enforce any additional one. If we end up changing the API I can take care of the changes.
Your call @LukasTy

Reply:
> Which makes me think that a different and maybe more explicit API could have been `yearsOrder?: 'asc' | 'desc'`. If at some point we change the default order, I guess we don't want people that use `reverseYears` to switch to `asc` 😆

Great suggestion. 👍 
However, I'm not sure if we'd ever need to change this default. 🤔 
I feel pretty safe going with a Boolean here. 👍

Reply:
This is a good suggestion. I had been thinking about how the terminology could be better and I think this is it. I can do it still, we've made it this far! :D

Reply:
Although I should mention that I had updated the docs to use "reverse chronological order" instead of ascending / descending, but if this seems more clear to everyone then we could use asc/desc. The "reverse chronological" seemed to be the easiest way to explain it's from most recent year to more distant. Can't really say "from newest to oldest".

If ascending / descending seems better to you both then I can update the api and terminology a bit still.

Reply:
To me, the current solution seems very clear and the DX is nice. 👌 
If you think that Flavien's suggestion sounds better—I'm not against it. 👍

Reply:
> However, I'm not sure if we'd ever need to change this default. 🤔

This is very unlikely indeed
The main benefit is that it's more explicit (`reverseYears`, but reverse from what order?)

But as I said, I clearly don't want to enforce it, this is not a super sensitive prop.

Reply:
> Although I should mention that I had updated the docs to use "reverse chronological order" instead of ascending / descending, but if this seems more clear to everyone then we could use asc/desc. The "reverse chronological" seemed to be the easiest way to explain it's from most recent year to more distant. Can't really say "from newest to oldest".

I used "asc" | "desc" but feel free to propose other values that are more time-based indeed.

Reply:
Since I have had the same question in my head about the explicitness ("reverse from what order"), I think this would indeed be much more clear and so worth doing.

Chronological and antichronological / counterchronological could be valid terms but I think asc and desc are most easily understandable as a fairly common convention.

Now I'm more familiar with the project this will be a piece of cake 🍰, this has been a good learning experience for me also, so perhaps I'll be able to make more contributions in the future.

Thanks y'all!